lib/rubocop/cop/style/date_time.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for uses of 'DateTime' that should be replaced by
+      # 'Date' or 'Time'.
+      #
+      # @example
+      #
+      #   # bad - uses DateTime for current time
+      #   DateTime.now
+      #
+      #   # good - uses Time for current time
+      #   Time.now
+      #
+      #   # bad - uses DateTime for modern date
+      #   DateTime.iso8601('2016-06-29')
+      #
+      #   # good - uses Date for modern date
+      #   Date.iso8601('2016-06-29')
+      #
+      #   # good - uses DateTime with start argument for historical date
+      #   DateTime.iso8601('1751-04-23', Date::ENGLAND
+      class DateTime < Cop
+        MSG = 'Prefer Date or Time over DateTime.'.freeze
+
+        def_node_matcher :uses_date_time?, <<-PATTERN
+          (send (const _ :DateTime) ...)
+        PATTERN
+
+        def_node_matcher :has_historic_date?, <<-PATTERN
```

Comment:
The cop that checks for extra prefixes is about to be extended to node matchers, and this will be an offense. 🙂 I would ✂️ both "uses" and "has" in this case.

Reply:
I'm sorry, but could explain what the offence would be? I'm happy to fix it, but I'm not quite sure what your referencing. I have all other issues patched except for that one.

Reply:
Prefixes such as `is_`, `has_`, `uses_`, etc. are omitted in idiomatic Ruby. There is a cop, `Naming/PredicateName` that checks for this in method definitions. We should follow the same convention when dynamically defining methods. 🙂